threestudio/data/uncond.py

```diff
@@ -82,42 +92,74 @@ def __init__(self, cfg: Any) -> None:
         ]
         self.height: int = self.heights[0]
         self.width: int = self.widths[0]
+        self.batch_size: int = self.batch_sizes[0]
         self.directions_unit_focal = self.directions_unit_focals[0]
+        self.elevation_range = self.cfg.elevation_range
+        self.azimuth_range = self.cfg.azimuth_range
+        self.camera_distance_range = self.cfg.camera_distance_range
+        self.fovy_range = self.cfg.fovy_range
+        self.step = -1
 
     def update_step(self, epoch: int, global_step: int, on_load_weights: bool = False):
         size_ind = bisect.bisect_right(self.resolution_milestones, global_step) - 1
         self.height = self.heights[size_ind]
         self.width = self.widths[size_ind]
+        self.batch_size = self.batch_sizes[size_ind]
         self.directions_unit_focal = self.directions_unit_focals[size_ind]
-        threestudio.debug(f"Training height: {self.height}, width: {self.width}")
+        threestudio.debug(
+            f"Training height: {self.height}, width: {self.width}, batch_size: {self.batch_size}"
+        )
+        # progressive view
+        self.progressive_view(global_step)
 
     def __iter__(self):
         while True:
             yield {}
 
+    def progressive_view(self, global_step):
+        r = min(1.0, global_step / (self.cfg.progressive_until + 1))
+        self.elevation_range = [
+            (1 - r) * self.cfg.eval_elevation_deg + r * self.cfg.elevation_range[0],
+            (1 - r) * self.cfg.eval_elevation_deg + r * self.cfg.elevation_range[1],
+        ]
+        self.azimuth_range = [
+            (1 - r) * 0.0 + r * self.cfg.azimuth_range[0],
+            (1 - r) * 0.0 + r * self.cfg.azimuth_range[1],
+        ]
+        self.camera_distance_range = [
+            (1 - r) * self.cfg.eval_camera_distance
+            + r * self.cfg.camera_distance_range[0],
+            (1 - r) * self.cfg.eval_camera_distance
+            + r * self.cfg.camera_distance_range[1],
+        ]
+        self.fovy_range = [
+            (1 - r) * self.cfg.eval_fovy_deg + r * self.cfg.fovy_range[0],
+            (1 - r) * self.cfg.eval_fovy_deg + r * self.cfg.fovy_range[1],
+        ]
```

Comment:
I recommend we don't vary the camera_distance_range and fovy_range progressively. Changing too many params at once makes it very hard to tune the hyperparams.

Reply:
Done!